src/blockrelay/blockrelay_common.cpp

```diff
@@ -280,3 +286,83 @@ void ThinTypeRelay::ClearBlockToReconstruct(CNode *pfrom)
     LOCK(cs_reconstruct);
     mapBlocksReconstruct.erase(pfrom->GetId());
 }
+
+bool ThinTypeRelay::ClearLargestBlockAndDisconnect(CNode *pfrom)
```

Comment:
I wonder if we could add a comment to explain a bit what this function is goign to do